### ISSUE TITLE
[FEAT]: Contact nginx reverse proxy settings and open source ID provider support

### ISSUE THREAD

**Issue by dalso0418** [azukaar/Cosmos-Server#40]
### Feature Description

Hello.

I run a home server directly in Korea and I am a very interested user.
Please understand even if there is a misunderstanding as it is a translation of Korean into English.

Cosmos is definitely a huge project and I would like to use it as a test, but before I use it, I would like to ask you a few questions about the existing ones to migrate from the version I am currently using.

Currently, we use Reverse Proxy using Nginx PRoxy Manager, and here we use ID providers such as Authentic to link SSO of sites.

Additionally, while working on a VPN project called netbird recently, I tried grpc_pass as well as generic proxy_pass for the first time, and I was wondering if you supported it.

To sum up, I wonder if it is possible to migrate the conf used by nginx.

ex)
authentik SSO Nginx Proxy Manager 설정 - https://www.geekscircuit.com/set-up-authentik-sso-with-nginx-proxy-manager/

netbird의 proxy 설정 - location /management {
grpc_pass grpc://192.168.1.18:443;
}

I'm looking forward to the future as I watch every update.
Thank you so much for the fantastic project :)

**Comment by azukaar** [comment]
안녕 !

>  I tried grpc_pass as well as generic proxy_pass for the first time, and I was wondering if you supported it.

Cosmos does not use NGINX so it does not "support" NGINX modules. On the other hand, Cosmos does support proxying request to other servers, including HTTP supersets such as WebDAV. Note that I have never tested GRPC.

It should work to create a URL such as:

```
{ 
  "Mode": "PROXY",
  "UsePathPrefix": true,
  "PathPrefix": "/management",
  "UseHost": true,
  "Host": "domain.com",
  "Target": "grpc://192.168.1.18:443"
}
```

But as I mentioned I have never actually tried it, so if you run into any issues surely we can figure it out together.

I will close this ticket for now, but feel free to comment if any other questions

**Comment by azukaar** [comment]
안녕 !

Cosmos does not use NGINX, so it does not "support" NGINX modules like the one you have linked.
On the other hand it can support similar features

- Yes you can create proxy routes to anything like you would with the proxy module. Simply create a URL of type "PROXY"

- Cosmos supports HTTP superset such as WebDav. I do not know if GRPC works, it should, but I have never really tested it. You can try it by creating a PROXY route to some GRPC services and see if it works. I will try it too later to see if I can get it to work, but as of now I cannot confirm if it really does

Thanks you! :)
